### PR TITLE
adding source-map-support for ts mapping in stack trace; enabling tsl…

### PR DESCRIPTION
**Pull request by andygreenegrass**
…int rules; added linting to travis

I added the source-map-support dev dependency which will resolve ts file and line numbers in stack traces during the tests.
I enabled some of the previously disabled tslint rules and made the updates to fix those.
I also added linting to travis.

### CODE REVIEW

**Comment by RishikeshDarandale** [comment]
How about using [husky](https://www.npmjs.com/package/husky) for linting and formatting at the time of commit?

**Comment by andygreenegrass** [comment]
I haven't used husky. I'll check it out. It _is_ completely unnecessary that linting runs for every node version in travis.

Also, can I rebase an active pull request? I usually rebase my commits before I push to github.

**Comment by andygreenegrass** [comment]
I added husky and lint-staged to do a partial lint of staged files on commit. If there are linting error it will attempt to fix what it can or fail to commit if the errors cannot be automatically fixed.

**Comment by jsdevel** [comment]
Absolutely no husky sorry.  Please remove that.

File: package.json [hunk 1]
```diff
@@ -60,13 +75,16 @@
     "express": "^4.16.4",
     "finalhandler": "^1.1.1",
     "glob": "^7.1.3",
+    "husky": "^1.3.1",
```
Comment (jsdevel) [inline package.json]: Please remove husky

Comment (RishikeshDarandale) [reply package.json]: @jsdevel , any particular reason to remove  `husky`?

**Comment by andygreenegrass** [comment]
husky is gone.

**Comment by jsdevel** [comment]
Thanks.  In a publicly shared repo, I'd like to limit what runs on my machine when making contributions.

**Comment by andygreenegrass** [comment]
I just went through and removed all unnecessary `self` variables. I wanted to get that in there along with the move to arrow functions. There will be no more changes for this pull request.

**Comment by andygreenegrass** [comment]
well, that's not good. I'm a little confused by the coverage results.

**Comment by andygreenegrass** [comment]
I think it's just because of the removed `const self = this;` lines.

**Comment by jsdevel** [comment]
i can accept the coverage decrease.  thanks @andygreenegrass !

**Pull request merged**